go/pkg/db/queries/key_update_credits_decrement.sql

```diff
@@ -0,0 +1,7 @@
+-- name: UpdateKeyCreditsDecrement :exec
+UPDATE `keys`
+SET remaining_requests = CASE 
+    WHEN remaining_requests - sqlc.arg('credits') > 0 THEN remaining_requests - sqlc.arg('credits')
```

Comment:
nit, we could do `remaining_requests >= sqlc.arg('credits')` instead

I think it's easier to read, but not really important